Platform agnostic role `resource_manager to manage network resources (#14)
ansible-collections/ansible.network: README.md

    junipernetworks.junos   2.2.0
    openvswitch.openvswitch 2.0.0
    vyos.vyos               2.3.0

> ## Using  platform-agnostic role ansible.network.resource_manager as part of ansible.network collection.**

Reviewer: ```suggestion
## Using  the platform-agnostic role ansible.network.resource_manager as part of ansible.network collection.**
```

> This role Provide a single platform agnostics entry point to manage all the resources supported for given network os.

Reviewer: ```suggestion
This role provides a single platform-agnostic entry point to manage all the resources supported for a given network OS.
```

**Capabilities**
```

> - Use list task to get the list of resource modules supported for a given network os.

Reviewer: ```suggestion
- Use list task to get the list of resource modules supported for a given network OS.
```

> - Use gather task to gather the facts of supported resource modules for a given network os.

Reviewer: ```suggestion
- Use gather task to gather the facts of supported resource modules for a given network OS.
```

> - Use configure task to perform a single operation (e.g merged) for a given network os.

Reviewer: ```suggestion
- Use configure task to perform a single operation (for example, merged) for a given network OS.
```

> - Use persis task to fetch the facts and save into inventory host vars files.

Reviewer: ```suggestion
- Use persist task to fetch the facts and save into inventory host vars files.
```

> - Use depoy task to push the saved inventory host vars on to the network appliance.

Reviewer: ```suggestion
- Use deploy task to push the saved inventory host vars on to the network appliance.
```

```
### Examples
##Using list task
Get the list of resource modules for given ansible_network_os
```yaml
run.yml
---
- hosts: ios
  gather_facts: no
  tasks:
  - name: invoke list function
    include_role:
      name: resource_manager
    vars:
      ansible_network_os: cisco.ios.ios
      perform_task: list
```
##Using gather task
Get the resource facts for provided resources for given ansible_network_os
```yaml
run.yml
---
- hosts: ios
  gather_facts: no
  tasks:

>   - name: invoke gather fuctnion

Reviewer: ```suggestion
  - name: invoke gather function
```

    include_role:
      name: resource_manager
    vars:
      perform_task: gather
      ansible_network_os: cisco.ios.ios
      resources:
        - 'interfaces'
        - 'l2_interfaces'
        - 'l3_interfaces'
```
##Using persist task
Get the resource facts for provided resources and build inventory host_vars for given ansible_network_os
```yaml
run.yml
---
- hosts: ios
  gather_facts: no
  tasks:

>   - name: invoke gather fuctnion

Reviewer: ```suggestion
  - name: invoke gather function
```

    include_role:
      name: resource_manager
    vars:
      perform_task: persist
      inventory_directory: ./inventory
      ansible_network_os: cisco.ios.ios
      resources:
        - 'interfaces'
        - 'l2_interfaces'
        - 'l3_interfaces'
```
##Using config task
Invoke single operation for provided resource with provided configuration and state for given ansible_network_os
```yaml
run.yml
---
- hosts: ios
  gather_facts: no
  tasks:

>   - name: invoke configure fuctnion

Reviewer: ```suggestion
  - name: invoke configure function
```

    include_role:
      name: resource_manager
    vars:
      ansible_network_os: cisco.ios.ios
      resource: interfaces
      perform_task: configure
      config:
        - name: "GigabitEthernet0/0"
          description: "Edited with Configure operation"
      state: merged
```
##Using deploy task
Perform ansible resource module supported operation with inventory host_vars as provided config for given ansible_network_os.
```yaml
run.yml
---
- hosts: ios
  gather_facts: no
  tasks:

>   - name: invoke deploy fuctnion

Reviewer: ```suggestion
  - name: invoke deploy function
```
